### ISSUE TITLE
Is this library not namespaced?

### ISSUE THREAD

**Issue by enkay** [segmentio/analytics-php#104]
Hi, am I missing something or is this library not namespaced?

**Comment by f2prateek** [comment]
Namespacing is not currently supported.

**Comment by f2prateek** [comment]
hey @enkay could you describe in more detail what issues you are seeing as a result of this?

**Comment by enkay** [comment]
TLDR: Namespacing is useful in larger projects to avoid naming conflicts and keep things organized.

I'm not using Segment at this time as it wasn't a good fit for other reasons, but considering Laravel is now more popular than Rails on Github, it would make sense to have an up to date PHP library, and it's unfortunate that no engineer at Segment is familiar with modern PHP development.

Here's some more information on namespacing:
http://php.net/manual/en/language.namespaces.php
https://stackoverflow.com/questions/5393108/whats-the-benefit-of-namespaces-in-php
https://stackoverflow.com/questions/9427674/php-namespacing-benefits

**Comment by f2prateek** [comment]
Thanks! Definitely agree we need help with the PHP library (https://segment.com/jobs/ 😉)

**Comment by f2prateek** [comment]
Triaged internally in JIRA https://segment.atlassian.net/browse/LIB-268

**Comment by enkay** [comment]
Hey guys it's been over a year and still nothing... I know Silicon Valley looks down on PHP, but is it really that hard to hire a PHP contractor for a couple weeks to fix or rewrite this library properly?

**Comment by dmyers** [comment]
~3 years later, is PHP not a focus for Segment at all?

**Comment by nd4p90x** [comment]
@dmyers @enkay My apologies for not responding sooner.  We are working on the PHP LIB and you can tell by our activities in the change log that new features are being added, issues are being resolved and releases are being made.   Currently we are working on adding PHP 8 support, along with a few other higher priority issues.   I agree with the namespace request and it is in our todo list.  My goal is to complete this after working through some current open issues.  Thank you.

**Comment by dmyers** [comment]
No worries @nd4p90x. Thanks for following up. That sounds great! Big fans of the product so very excited for future releases.

**Comment by mischasigtermans** [comment]
Will this ever happen?
